docs/source/extension/plugin_system.rst

```diff
@@ -0,0 +1,96 @@
+.. Copyright (c) Jupyter Development Team.
+.. Distributed under the terms of the Modified BSD License.
+
+The JupyterLab Plugin System
```

Comment:
Wondering if this deserves a new page or if it could go in https://jupyterlab.readthedocs.io/en/latest/extension/extension_dev.html ?

Reply:
Or maybe we can move the current content of https://jupyterlab.readthedocs.io/en/latest/extension/extension_dev.html to this new `plugin_system.rst`, so it shows up in the side bar as "The JupyterLab Plugin System":

![image](https://github.com/jupyterlab/jupyterlab/assets/591645/03b618c3-58a8-43ed-a4f1-e34a33f50540)

Reply:
This is a good question. I started this document because I thought there was a gap between the very high level explanation offered in extension_dev and the technical steps and understanding someone needs to use it in the code. Maybe it should be integrated and the page renamed, which could be beneficial...though it would take some refactoring of the main document for sure. I'll take a look.

Reply:
I decided to keep this document separate so that it functions as a focused high level design document (with an example). The main page can use some refactoring, maybe as a future PR...